pkg/virt-handler/selinux/context_executor.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package selinux
+
+import (
+	"fmt"
+	"os/exec"
+	"runtime"
+	"syscall"
+
+	"github.com/opencontainers/selinux/go-selinux"
+)
+
+type SELinuxContextExecutor struct {
+	cmdToExecute  *exec.Cmd
+	desiredLabel  string
+	originalLabel string
+	pid           int
+}
+
+func NewSELinuxContextExecutor(pid int, cmd *exec.Cmd) (*SELinuxContextExecutor, error) {
+	desiredLabel, err := getSelinuxLabelForPID(pid)
+	if err != nil {
+		return nil, err
+	}
+	originalLabel, err := getOriginalSelinuxLabel()
+	if err != nil {
+		return nil, err
+	}
+	return &SELinuxContextExecutor{
+		pid:           pid,
+		cmdToExecute:  cmd,
+		desiredLabel:  desiredLabel,
+		originalLabel: originalLabel,
+	}, nil
+}
+
+func (ce SELinuxContextExecutor) Execute() error {
+	if isSELinuxEnabled() {
```

Comment:
> No reason to; is a utility function that doesn't require any state encapsulated in this struct; let's leave this one out.

In case this is executed multiple times, there is no need to re-check it on each execution.
But this is a nit, especially that it is probably not used more than once. Not going to insist on this, leaving it up to you.